front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3Support.java

```diff
@@ -113,6 +113,30 @@ public T findById(String id) throws NotFoundException {
     }
   }
 
+  public Collection<T> allVersionsOf(String id) throws NotFoundException {
+    try {
+      VersionListing versionListing = amazonS3.listVersions(new ListVersionsRequest(bucket, buildS3Key(id), null, null, null, 20));
```

Comment:
ah - I see 20 maxResults here - maybe make that configurable (and / or support it as a request option)